runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/MetricsToCounterUpdateConverter.java

```diff
@@ -110,6 +111,16 @@ public static CounterUpdate fromStringSet(MetricKey key, StringSetData stringSet
         .setStringList(stringList);
   }
 
+  public static CounterUpdate fromBoundedTrie(MetricKey key, BoundedTrieData boundedTrieData) {
+    // BoundedTrie uses SET kind metric aggregation which tracks unique strings.
+    CounterStructuredNameAndMetadata name = structuredNameAndMetadata(key, Kind.SET);
+    // TODO (rosinha): Once the CounterUpdate API is updated in dataflow client update this.
+    return new CounterUpdate()
+        .setStructuredNameAndMetadata(name)
+        .setCumulative(false)
+        .set("bounded_trie", boundedTrieData.toProto());
```

Comment:
Will the dataflow worker even accept this yet? Again, let's pull the legacy worker changes out into their own PR.